integration/clean/clean_aws_resource.go

```diff
@@ -0,0 +1,145 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build clean
+// +build clean
+
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2Type "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	ssmType "github.com/aws/aws-sdk-go-v2/service/ssm/types"
+	smithyTime "github.com/aws/smithy-go/time"
+)
+
+const (
+	daysToKeep   = 30
+	keepDuration = -1 * time.Hour * 24 * time.Duration(daysToKeep)
+)
+
+var expirationDate = time.Now().UTC().Add(keepDuration)
+
+func main() {
+	log.Println("Begin to clean EC2 AMI")
+	cleanAMI()
+
+	log.Println("Begin to clean SSM Parameter Store")
+	cleanSSMParameterStore()
+
+	log.Println("Finished cleaning resources.")
+}
```

Comment:
On the one hand, I see value in cleanly exiting successfully always. On the other hand, what this does is it swallows issues that we would never notice because the job always succeeds. Without a notification in place, this doesn't matter either way. 

Personally, my opinion is that we should retain the existing behavior where we failed the job when there is an error, and we can revisit that when we work towards implementing some notification mechanism so we know that there is an issue with the job and someone needs to fix it.

Reply:
I prefer the former since it would be good for cleaning all the resources without having to hold back. However, if we prefer to fix ASAP when the errors occur, I'm good with that too.

Reply:
Can you elaborate? What I'm saying is that regardless of errors or not, currently (even with this change), we are slow to react to issues with this GitHub action. By making it succeed even when there are errors with deleting resources, it makes it harder for us to tell that something is wrong.

Reply:
To be more specific, failing a job early would  make us tackle handle errors frequently. However, the resources for release's integration test are not created frequently, only in a short amount of time and the release happens every 3 months. On the other hand, if we shallow the errors, they would shrimp down the successes `deleted resources` and during the next month, we only need to encounter the errors though. So the question was: do we want to tackle the errors only once a month(infrequently) or tacking the errors immediately? Giving the circumstances of release one a month and shrimping down the resources to delete one a month and tackle only that time, I would prefer the *shallow errors* way.  However, if we prefer to tackle immediately, I am good by that too.